torch/_inductor/ir.py

```diff
@@ -6564,6 +6564,8 @@ class WhileLoop(ExternKernel):
     cond_subgraph: Optional[Subgraph] = None
     body_subgraph: Optional[Subgraph] = None
     outputs: Optional[List[MultiOutput]] = None
+    mutated_inputs: Optional[List[TensorBox]] = None
+    iter_idx_expr: Optional[sympy.Expr] = None
```

Comment:
Add some documentation explaining what `iter_idx_expr` is. I don't quite understand why this needs to be separate from the rest of the body/cond?

Reply:
Sure. The basic idea is that subgraph will bake in the symbol in the IR nodes and it needs to be assigned and incremented in the wrapper code as the loop index. I can write a more detailed description for this.

Reply:
+ 1, don't follow.

If you mean number of iterations - that is typically given as `trip_count`

Reply:
For scan, the trip_count can be a SymInt that depends on the shape of input tensors so we need to use a symbolic expression for codegen. Any suggestions on how to do this better?